### ISSUE TITLE
What's the machine requirements for each model?

### ISSUE THREAD

**Issue by diegofer25** [meta-llama/codellama#30]
I want to know what's the minimum requirement memory/CPU/GPU for each model to run relatively fast. I ran in my M1

```
torchrun --nproc_per_node 1 example_completion.py \
    --ckpt_dir CodeLlama-7b/ \
    --tokenizer_path CodeLlama-7b/tokenizer.model \
    --max_seq_len 128 --max_batch_size 4
NOTE: Redirects are currently not supported in Windows or MacOs.
> initializing model parallel with size 1
> initializing ddp with size 1
> initializing pipeline with size 1
Loaded in 182.59 seconds
```

and it's taking more than 5 minutes.

**Comment by GaganHonor** [comment]
Well ,the 7B model can be served on a single GPU, while the 34B model returns the best results and allows for better coding assistance, but the smaller 7B and 13B models are faster and more suitable for tasks that require low latency, like real-time code completion.  for a 7B parameter model, you need about 14GB of RAM to run it in float16 precision. and 🧑‍💻It’s important to note that running CodeLlama on an M1 Mac may not be optimal as it is not officially supported. The M1 chip is based on ARM architecture, which is different from the x86 architecture used by most GPUs and CPUs. This can result in compatibility issues and slower performance when running software that is not optimized for ARM 🙏 Thanks a lot for reading

**Comment by Hangsiin** [comment]
I could run 7B  model on google Colab environment with T4 GPU (Free GPU). 
7B model is pretty light to use.
But it takes many system RAM when loading so colab pro access is required for more RAM.
It took times about 3~4 minutes for loading.
But Inference was quiet fast!
I was surprised to find that I could run LLM with a single T4 GPU.

**Comment by GaganHonor** [comment]
Yes , I tested 7B model on AWS EC2 Instance t2.micro , t3a.micro , and it worked quite well there are 
many more free GPU also there please check it out , and for Trying out the model i find https://huggingface.co/blog/codellama#conversational-instructions very useful 🙏

**Comment by MattOates** [comment]
> It’s important to note that running CodeLlama on an M1 Mac may not be optimal as it is not officially supported. The M1 chip is based on ARM architecture, which is different from the x86 architecture used by most GPUs and CPUs. This can result in compatibility issues and slower performance when running software that is not optimized for ARM

The model is just data, with llama.cpp the models run at realtime speeds with Metal acceleration on M1/2. Thanks to unified memory of the platform if you have 32GB of RAM that's all available to the GPU. Its a dream architecture for running these models, why would you put anyone off? My laptop on battery power can run 13b llama no trouble.
